src/PriceFeed.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/// @title PriceFeed
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+pragma solidity ^0.8.15;
+
+import { IPriceFeed } from './IPriceFeed.sol';
+import { AggregatorV3Interface } from './AggregatorV3Interface.sol';
+import { SafeCast } from 'openzeppelin-contracts/contracts/utils/math/SafeCast.sol';
+
+contract PriceFeed is IPriceFeed {
+    using SafeCast for int256;
+
+    uint256 constant WAD_DECIMALS = 18;
+
+    error StaleOracle(uint256 updatedAt);
+
+    AggregatorV3Interface public immutable chainlink;
+    uint8 public immutable decimals;
+    uint256 public immutable staleAfter;
+    uint256 public immutable decimalFactor;
+
+    constructor(AggregatorV3Interface _chainlink, uint256 _staleAfter) {
+        chainlink = _chainlink;
+        decimals = chainlink.decimals();
+        staleAfter = _staleAfter;
+
+        uint256 decimalFactorTemp = 1;
+        if (decimals < WAD_DECIMALS) {
+            decimalFactorTemp = 10**(WAD_DECIMALS - decimals);
```

Comment:
chainlink feeds are always 18 for feeds quoted against ETH and 8 for others
see: https://docs.chain.link/docs/ethereum-addresses/

maybe we can avoid the decimal calculations based on that assumption

Reply:
have you found an official place in their docs where they say this?

Reply:
Chainlink docs are not amazing :)